AK/Base64.cpp

```diff
@@ -140,4 +142,100 @@ ErrorOr<String> encode_base64(ReadonlyBytes input)
     return output.to_string();
 }
 
+// https://infra.spec.whatwg.org/#forgiving-base64
+ErrorOr<ByteBuffer> decode_forgiving_base64(StringView input)
+{
+    // 1. Remove all ASCII whitespace from data.
+    auto data = input.trim_whitespace();
```

Comment:
`trim_whitespace` does _not_ remove all ASCII whitespace from `data`, it simply removes it from the beginning and/or end. This means an (albeit silly) input such as `dG   Vz dA  o=   ` will fail when it shouldn't.